include/open62541/server_pubsub.h

```diff
@@ -925,6 +927,41 @@ UA_Server_addSecurityGroup(UA_Server *server, UA_NodeId securityGroupFolderNodeI
 UA_StatusCode UA_EXPORT UA_THREADSAFE
 UA_Server_removeSecurityGroup(UA_Server *server, const UA_NodeId securityGroup);
 
+/**
+ * @brief This is a repeated callback which is triggered on each iteration of SKS Pull request.
+ * The server uses this callback to notify user about the status of current Pull request iteration.
+ * The period is calculated based on the KeylifeTime of specified in the SecurityGroup object node on
+ * the SKS server.
+ *
+ * @param server The server instance managing the publisher/subscriber.
+ * @param sksPullRequestStatus The current status of sks pull request.
+ * @param userData The pointer to user defined data passed to this callback.
+ */
+typedef void
+(*UA_Server_sksPullRequestCallback)(UA_Server *server, UA_StatusCode sksPullRequestStatus, void* userData);
+
+/**
+ * @brief Sets the SKS client config used to call the GetSecurityKeys Method on SKS and get the
+ * initial set of keys for a SecurityGroupId and adds timedCallback for the next GetSecurityKeys
+ * method Call. This uses async Client API for SKS Pull request
+ *
+ * @note It is be called before setting Reader/Writer Group into Operational because this also allocates
+ * a channel context for the pubsub security policy.
+ *
+ * @note the stateCallback of sksClientConfig will be overwritten by an internal callback.
+ *
+ * @param server the server instance
+ * @param sksClientConfig holds the required configuration to make encrypted connection with SKS Server
+ * @param sksEndpointUrl holds the endpointUrl of the SKS server
+ * @param securityGroupId the SecurityGroupId of the securityGroup on SKS and reader/writergroups
+ * @param callback the user defined callback to notify the user about the status of SKS Pull request.
+ * @param userData passed to the callback function
+ * @return UA_StatusCode the retuned status
+ */
+UA_StatusCode UA_EXPORT
+UA_Server_setSksClient(UA_Server *server, UA_ClientConfig *sksClientConfig, const char * sksEndpointUrl, UA_String securityGroupId,
```

Comment:
Please move the "securityGroupId" to be the second argument.
We typically first indicate the context or scope of the operation (here server->securitygroup) and then the arguments of the operation.